### PR TITLE
[SYCLomatic] Add CMake migration rule, lit test for cuda_compile_fatbin

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**

### CODE REVIEW

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml [hunk 1]
```diff
@@ -264,3 +264,16 @@
   CmakeSyntax: add_executable_opts
   In: add_executable(${value} OPTIONS ${options})
   Out: add_executable(${value})
+
+- Rule: rule_cuda_compile_fatbin
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: cuda_compile_fatbin
+  In: cuda_compile_fatbin(${value})
```
Comment (tomflinda) [inline clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml:515]: According to the definition of cuda_compile_fatbin:
```
cuda_compile_fatbin(<generated_files> <file>... [OPTIONS ...])
```
The "[OPTIONS ...]" should also be considered, pls add a new yaml rule to cover this scenarios, like Line 264 does.

Comment (daiyaan-ahmed6) [reply clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml:515]: Done.

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml [hunk 2]
```diff
@@ -264,3 +264,23 @@
...
+    value:
+      In: ${arg}.cu
+      Out: ${arg}.dp.cpp
+      MatchMode: Partial
+      RuleId: "replace_source_extension_with_dp_cpp"
+
+- Rule: rule_dpct_helper_compile_sycl_code_remove_opts
+  Kind: CMakeRule
+  Priority: Fallback
+  CmakeSyntax: dpct_helper_compile_sycl_code_remove_opts
```
Comment (tomflinda) [inline clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml]:
```suggestion
  cuda_compile_fatbin_opts
```

Comment (daiyaan-ahmed6) [reply clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml]: Done.

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml [hunk 3]
```diff
@@ -264,3 +264,23 @@
...
+  In: cuda_compile_fatbin(${value})
+  Out: dpct_helper_compile_sycl_code(${value})
+  Subrules:
+    value:
+      In: ${arg}.cu
+      Out: ${arg}.dp.cpp
+      MatchMode: Partial
+      RuleId: "replace_source_extension_with_dp_cpp"
+
+- Rule: rule_dpct_helper_compile_sycl_code_remove_opts
```
Comment (tomflinda) [inline clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml]:
```suggestion
- Rule: rule_cuda_compile_fatbin_remove_opts
```

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml (hunk 3 above)
Comment (daiyaan-ahmed6) [reply clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml]: Done.

**Review by intwanghao — APPROVED** [review]
LGTM

**Pull request merged**